src/klein/app.py

```diff
@@ -302,7 +304,10 @@ def error_handler(request, failure):
         def deco(f):
             @wraps(f)
             def _f(instance, request, failure):
-                return _call(instance, f, request, failure)
+                r = _call(instance, f, request, failure)
+                if IRenderable.providedBy(r):
```

Comment:
I think this is duplicating this function https://github.com/twisted/klein/blob/3d17cb1abb0c16de4a55998f0c02812dd7a6283f/src/klein/resource.py#L219-L234 presently defined in the middle of `KleinResource.render`.  I think the desired functionality here is probably to have all the same automatical adaptifying that happens with a value returned from a regular renderer?